### PR TITLE
Adapt Razor runtime compilation guidance to changes made in ASP.NET Core 3.1

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/17731
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/17770

Adds a new moniker section for ASP.NET Core 3.1+ and removes unused image and code files.

[Internal review page (ASP.NET Core 3.1+)](https://review.docs.microsoft.com/aspnet/core/mvc/views/view-compilation?view=aspnetcore-3.1&branch=pr-en-us-17777)

### CODE REVIEW

File: aspnetcore/mvc/views/view-compilation.md [hunk 1]
```diff
@@ -4,14 +4,91 @@ author: rick-anderson
...
+1. Install the [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation/) NuGet package.
+1. Modify the launch profile `environmentVariables` section in *launchSettings.json*:
+    * Set the `ASPNETCORE_ENVIRONMENT` environment variable to `Development`.
+    * Set the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable to the runtime compilation assembly name.
+
+In the following example, runtime compilation is enabled in the Development environment for the `IIS Express` and `RazorPagesApp` launch profiles:
+
+[!code-json[](~/mvc/views/view-compilation/samples/3.1/launchSettings.json?highlight=15-16,24-25)]
+
+No code is needed in the project's `Startup` class. At runtime, ASP.NET Core searches for an [assembly-level HostingStartup attribute](xref:fundamentals/configuration/platform-specific-configuration#hostingstartup-attribute) in `Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation`&mdash;the assembly specified by the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable. The `HostingStartup` attribute specifies the app startup code to execute. That startup code enables runtime compilation.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/view-compilation.md:79]:
```suggestion
No code changes are needed in the project's `Startup` class. At runtime, ASP.NET Core searches for an [assembly-level HostingStartup attribute](xref:fundamentals/configuration/platform-specific-configuration#hostingstartup-attribute) in `Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation`&mdash;the assembly specified by the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable. The `HostingStartup` attribute specifies the app startup code to execute. That startup code enables runtime compilation.
```

Comment (Rick-Anderson) [inline aspnetcore/mvc/views/view-compilation.md:79]: per style guide, avoid mdash
plus the sentence is too long to MT
It's easy to split into two sentences, but I'm not sure you need specified by the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable

File: aspnetcore/mvc/views/view-compilation.md [hunk 2]
```diff
@@ -4,14 +4,91 @@ author: rick-anderson
...
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddRazorPages()
+            .AddRazorRuntimeCompilation();
+
+        // code omitted for brevity
+    }
+    ```
+
+## Conditionally enable runtime compilation in existing project
```
[the remark this answers is not in the document]
Comment (scottaddie) [reply aspnetcore/mvc/views/view-compilation.md]: Nice suggestions. Thank you for taking a look!

File: aspnetcore/mvc/views/view-compilation.md [hunk 3]
```diff
@@ -4,14 +4,91 @@ author: rick-anderson
...
+        // code omitted for brevity
+    }
+    ```
+
+## Conditionally enable runtime compilation in an existing project
+
+Runtime compilation can be enabled such that it's only available for local development. Conditionally enabling in this manner ensures that the published output:
+
+* Uses compiled views.
+* Is smaller in size.
```
Comment (pranavkm) [inline aspnetcore/mvc/views/view-compilation.md:65]: I would remove this line. It was the one side-effect we couldn't fix with this implementation

**Comment by scottaddie** [comment]
Thank you @serpent5, @Rick-Anderson, and @pranavkm for the quick review! 👍

**Pull request merged**